src/zarr/api/asynchronous.py

```diff
@@ -0,0 +1,925 @@
+from __future__ import annotations
+
+import asyncio
+import warnings
+from collections.abc import Iterable
+from typing import Any, Literal, Union, cast
+
+import numpy as np
+import numpy.typing as npt
+
+from zarr.abc.codec import Codec
+from zarr.array import Array, AsyncArray
+from zarr.buffer import NDArrayLike
+from zarr.chunk_key_encodings import ChunkKeyEncoding
+from zarr.common import JSON, MEMORY_ORDER, ChunkCoords, OpenMode, ZarrFormat
+from zarr.group import AsyncGroup
+from zarr.metadata import ArrayV2Metadata, ArrayV3Metadata
+from zarr.store import (
+    StoreLike,
+    make_store_path,
+)
+
+ShapeLike = tuple[int, ...]  # TODO: support int for shape
```

Comment:
Can we use `ChunkCoords` here?

Reply:
yes, but maybe not for long. I went ahead and changed it but ShapeLike was intended to be `int | tuple[int, ...]`... Perhaps that will work for ChunkCoords too?